packages/mdc-menu/simple/foundation.js

```diff
@@ -91,8 +91,10 @@ export default class MDCSimpleMenuFoundation extends MDCFoundation {
     this.keyupHandler_ = (evt) => this.handleKeyboardUp_(evt);
     /** @private {function(!Event)} */
     this.documentClickHandler_ = (evt) => {
-      this.adapter_.notifyCancel();
-      this.close(evt);
+      if (!evt.target.classList.contains('mdc-list-item')) {
```

Comment:
Basically, but it might be better done as a loop to avoid recursive calls?  (Also your code would either return `true` or `null` as written.)

```js
while (el) {
  if (el.classList.contains(className)) {
    return true;
  }
  el = el.parentNode;
}
return false;
```